test/unit/fact_parser_test.rb

```diff
@@ -297,6 +297,22 @@ class FactParserTest < ActiveSupport::TestCase
       assert_equal '00:00:00:00:00:12', found.last[:macaddress]
     end
 
+    test "#suggested_primary_interface detects primary interface using DNS IPv6" do
+      parser.stubs(:interfaces).returns({
+        'br0' => {'ipaddress6' => '2001:db8::10', 'macaddress' => '00:00:00:00:00:10'},
+        'em1' => {'ipaddress6' => '2001:db8::20', 'macaddress' => '00:00:00:00:00:20'},
+        'em2' => {'ipaddress6' => '2001:db8::30', 'macaddress' => '00:00:00:00:00:30'},
+        'bond0' => {'ipaddress6' => '2001:db8::40', 'macaddress' => '00:00:00:00:00:40'},
+      }.with_indifferent_access)
```

Comment:
I'd prefer to be stricter and provide just a regular hash with the right types. Though you are right that it's presented as indifferent access
```suggestion
        'br0' => {ipaddress6: '2001:db8::10', macaddress: '00:00:00:00:00:10'},
        'em1' => {ipaddress6: '2001:db8::20', macaddress: '00:00:00:00:00:20'},
        'em2' => {ipaddress6: '2001:db8::30', macaddress: '00:00:00:00:00:30'},
        'bond0' => {ipaddress6: '2001:db8::40', macaddress: '00:00:00:00:00:40'},
      })
```

Reply:
I prefer having the style consistent with the rest of the file. So unless you feel it's really wrong, I think we should leave it with the indifferent access.